packages/fxa-content-server/app/scripts/views/subscriptions_product_redirect.js

```diff
@@ -15,31 +15,43 @@ class SubscriptionsProductRedirectView extends FormView {
   initialize(options) {
     this._currentPage = options.currentPage;
     this._productId = this._currentPage.split('/').pop();
-    this._queryParams = Url.searchParams(this.window.location.search);
     this.relier.set('subscriptionProductId', this._productId);
 
     this._subscriptionsConfig = {};
     if (options && options.config && options.config.subscriptions) {
       this._subscriptionsConfig = options.config.subscriptions;
     }
 
-    this.mustAuth =
-      !!this._queryParams.signin ||
-      !this._subscriptionsConfig.allowUnauthenticatedRedirects;
-
     // Flow events need to be initialized before the navigation
     // so the flow_id and flow_begin_time are propagated
     this.initializeFlowEvents();
   }
 
+  render() {
+    const account = this.getSignedInAccount();
+    return Promise.resolve()
+      .then(() => {
+        const isSignedInPromise =
+          account?.isSignedIn() || Promise.resolve(false);
+        isSignedInPromise.then((isSignedIn) => {
+          this._redirectPath =
+            !isSignedIn &&
+            this._subscriptionsConfig.allowUnauthenticatedRedirects
+              ? `checkout/${this._productId}`
+              : `products/${this._productId}`;
```

Comment:
This is still based on the feature flag, so that we don't automatically redirect unauthned users to the new checkout route on payments in prod until it's ready.